tests/components/manual_mqtt/test_alarm_control_panel.py

```diff
@@ -11,1772 +10,1698 @@
-                    "state_topic": "alarm/state",
-                    "command_topic": "alarm/command",
-                    "payload_arm_away": "ARM_AWAY",
-                }
-            },
-        )
-        self.hass.block_till_done()
-
-        entity_id = "alarm_control_panel.test"
-
-        assert STATE_ALARM_DISARMED == self.hass.states.get(entity_id).state
-
-        # Fire the arm command via MQTT; ensure state changes to pending
-        fire_mqtt_message(self.hass, "alarm/command", "ARM_AWAY")
-        self.hass.block_till_done()
-        assert STATE_ALARM_PENDING == self.hass.states.get(entity_id).state
-
-        # Fast-forward a little bit
-        future = dt_util.utcnow() + timedelta(seconds=1)
-        with patch(
-            (
-                "homeassistant.components.manual_mqtt.alarm_control_panel."
-                "dt_util.utcnow"
-            ),
-            return_value=future,
-        ):
-            fire_time_changed(self.hass, future)
-            self.hass.block_till_done()
-
-        assert STATE_ALARM_ARMED_AWAY == self.hass.states.get(entity_id).state
-
-    def test_arm_night_via_command_topic(self):
-        """Test arming night via command topic."""
-        assert setup_component(
-            self.hass,
-            alarm_control_panel.DOMAIN,
-            {
-                alarm_control_panel.DOMAIN: {
-                    "platform": "manual_mqtt",
-                    "name": "test",
-                    "pending_time": 1,
-                    "state_topic": "alarm/state",
-                    "command_topic": "alarm/command",
-                    "payload_arm_night": "ARM_NIGHT",
-                }
-            },
-        )
-        self.hass.block_till_done()
-
-        entity_id = "alarm_control_panel.test"
-
-        assert STATE_ALARM_DISARMED == self.hass.states.get(entity_id).state
-
-        # Fire the arm command via MQTT; ensure state changes to pending
-        fire_mqtt_message(self.hass, "alarm/command", "ARM_NIGHT")
-        self.hass.block_till_done()
-        assert STATE_ALARM_PENDING == self.hass.states.get(entity_id).state
-
-        # Fast-forward a little bit
-        future = dt_util.utcnow() + timedelta(seconds=1)
-        with patch(
-            (
-                "homeassistant.components.manual_mqtt.alarm_control_panel."
-                "dt_util.utcnow"
-            ),
-            return_value=future,
-        ):
-            fire_time_changed(self.hass, future)
-            self.hass.block_till_done()
-
-        assert STATE_ALARM_ARMED_NIGHT == self.hass.states.get(entity_id).state
-
-    def test_disarm_pending_via_command_topic(self):
-        """Test disarming pending alarm via command topic."""
-        assert setup_component(
-            self.hass,
+async def test_fail_setup_without_state_topic(hass, mqtt_mock):
+    """Test for failing with no state topic."""
+    with assert_setup_component(0) as config:
+        assert await async_setup_component(
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5745133)